auth_user_case_insensitive/models/res_users.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+
+    _inherit = 'res.users'
+
+    login = fields.Char(
+        help='Used to log into the system. Case insensitive.',
+    )
+
+    @api.model
+    def search(self, domain, *args, **kwargs):
+        """ Overload search to lowercase name domains. This can't be done in
+        a regular search method as the field is not computed
+        """
+        for idx, _domain in enumerate(domain):
+            if _domain[0] == 'login':
+                lower = _domain[2].lower() if _domain[2] else False
+                domain[idx] = (_domain[0], _domain[1], lower)
+        return super(ResUsers, self).search(domain, *args, **kwargs)
```

Comment:
True, but when we have case insensivity, an user expects to be able to login typing wither "eric", "Eric" or "ERIC".
To avoid hacking search(), we can just extend `_login` to lowercase the login before calling super(),